### PR TITLE
Ofir perturhska/issue 33 php "external" mode issue - tests

### PR DESCRIPTION
**Pull request by ofir-petrushka**
Sorry it's a bit all in one, please review by commit.

Relates to issue https://github.com/chr4-cookbooks/newrelic-ng/issues/33

* Add  a test for the issue in question check by kitchen as:
bundle exec kitchen test php-external-default-ubuntu-12
(please don't ask why we still use ubuntu-12.04)
* Updates .rubocop
* Fixes to comply with current rubocop
* Travis-ci rubocop test integration. (probably needs a few key presses on your side to get right, login&enable repo) 
* Fixes to mini tests for 'external' php mode

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+nguage: ruby
```
Comment (chr4) [inline .travis.yml]: `nguage` Is this a typo?

File: attributes/generic-agent.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 default['newrelic-ng']['generic-agent']['ruby-packages'] = value_for_platform_family(
   'rhel' => %w(ruby ruby-devel rubygems),
-  'debian' => %w(ruby ruby-dev),
+  'debian' => %w(ruby ruby-dev)
```
Comment (chr4) [inline attributes/generic-agent.rb]: `TrailingCommaInLiteral` should complain here, shouldn't it?

File: .ruby-version [hunk 3]
```diff
@@ -0,0 +1 @@
+2.2.4
```
Comment (chr4) [inline .ruby-version]: Was there a specific reason why you commited `.ruby-version`? To my knowledge this is not a best practise. Please remove this commit.

File: attributes/generic-agent.rb (hunk 2 above)
Comment (ofir-petrushka) [inline attributes/generic-agent.rb]: They seem to have separated TrailingCommaInLiteral, TrailingCommaInArguments.
Adding it.

File: .ruby-version (hunk 3 above)
Comment (ofir-petrushka) [inline .ruby-version]: pin vs unpin, depends if your pro latest or pro stable.
your repo = your way.
fixing.

**Comment by ofir-petrushka** [comment]
@chr4 please review again.

**Comment by chr4** [comment]
About `.ruby-version`, I usually agree of putting it into ruby project, but as `chef-client` doesn't respect it anyway, I think it's a good practise to not commit it into chef cookbooks.

File: .gitignore [hunk 4]
```diff
@@ -1,6 +1,10 @@
+# don't pin versions
```
Comment (chr4) [inline .gitignore]: I don't think comments in `.gitignore` are necessary. Would you mind removing them?

File: attributes/nrsysmond.rb [hunk 5]
```diff
@@ -34,9 +34,9 @@
 }
 
 default['newrelic-ng']['nrsysmond']['config_file'] = '/etc/newrelic/nrsysmond.cfg'
-default['newrelic-ng']['nrsysmond']['mode'] = 00640
+default['newrelic-ng']['nrsysmond']['mode'] = 0o640
```
Comment (chr4) [inline attributes/nrsysmond.rb]: This looks like a typo :)

**Comment by chr4** [comment]
Besides the comment in `.gitignore` this looks good to me! Thanks for your contribution!
I'll release this once the comment is removed and the tests run fine.

Would you mind being added to the contributors list?

**Comment by ofir-petrushka** [comment]
* .gitignore comments removed.

* about the .ruby-version originally was for bundler. 
  I work a lot currently with the "pin everything" for stability, so all lock files, all packages etc.. (personally I like use latest, but you know...)
  Since we have a policy for pining everything, I didn't have a default ruby version, so when the error came that was the 1st path.
  Travis is also listening to it, and if it's there you don't need to put it in the .travis.yml. (not stating it defaults to ruby 1.9)
  https://github.com/chr4-cookbooks/newrelic-ng/pull/34/files#diff-354f30a63fb0907d4ad57269548329e3R3

* the 0o640 instead of 00640 is explicitly required by the latest rubocop. 
  I assume other formats are acceptable, and probably can configure rubocop otherwise but that was "least changes" to be compliant. 

* I am ok with being added to the contributors list.

**Comment by chr4** [comment]
> the 0o640 instead of 00640 is explicitly required by the latest rubocop.
I assume other formats are acceptable, and probably can configure rubocop otherwise but that was "least changes" to be compliant.

I just realized that. Didn't know that before, but it makes sense. Merging. Thanks!

**Comment by chr4** [comment]
Released in `v0.6.0`.

**Pull request merged**